network/src/logging.rs

```diff
@@ -69,4 +70,33 @@ impl<'a> NetworkSchema<'a> {
             .remote_peer(&metadata.remote_peer_id)
             .network_address(&metadata.addr)
     }
+
+    // Increments a counter, and logs with the network context.
+    pub fn with_count(self, counter: &IntCounterVec, amount: i64) -> Self {
+        counter
+            .with_label_values(&[
+                self.network_context.role().as_str(),
+                self.network_context.network_id().as_str(),
+                self.network_context.peer_id().short_str().as_str(),
+            ])
+            .inc_by(amount);
+        self
+    }
```

Comment:
this seems kind of awkward to me... tbh the problem with these metrics is taking an array of `&str`, which makes it kind of hard to compose functions that just add context

Reply:
It's awkward, and very specific to networking.  I was just exploring ways to integrate counters into the logs.

Many people have been asking for counters integrated with logs